ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -71,44 +74,96 @@ class MainActivity : ComponentActivity() {
+        // IMPORTANT NOTE: Usually you want to show the vignette all the time on screens with
+        // scrolling content, a rolling side button, or a rotating bezel. This preference is just
+        // to visually demonstrate the vignette for the developer to see it on and off.
+        var vignetteVisiblePreference by rememberSaveable { mutableStateOf(true) }
+
+        // Observes the current back stack entry to show the vignette for the appropriate screens by
+        // checking the destination + route.
+        // TODO: Replace with .currentBackStackEntryAsState() once added to Compose for Wear OS
+        // Navigation Library: https://issuetracker.google.com/issues/212739653
+        val currentBackStackEntry by swipeDismissableNavController
+            .currentBackStackEntryFlow.collectAsState(null)
+
+        val currentRoute = currentBackStackEntry?.destination?.route
 
         Scaffold(
             // Scaffold places time at top of screen to follow Material Design guidelines.
             timeText = {
+                val activelyScrolling =
+                    scrollStates[currentScrollStateKey]?.isScrollInProgress ?: false
+
                 CustomTimeText(
-                    visible = !scalingLazyListState.isScrollInProgress,
+                    visible = !activelyScrolling,
                     showLeadingText = showProceedingTextBeforeTime,
                     leadingText = stringResource(R.string.leading_time_text)
                 )
             },
             vignette = {
-                CustomVignette(
-                    visible = vignetteVisible,
-                    vignettePosition = VignettePosition.TopAndBottom
-                )
+                // Only show vignette for screens with scrollable content.
+                currentRoute?.let {
+                    if (currentRoute.startsWith(Screen.WatchList.route)) {
+                        CustomVignette(
+                            visible = vignetteVisiblePreference,
+                            vignettePosition = VignettePosition.TopAndBottom
+                        )
+                    } else if (currentRoute.startsWith(Screen.WatchDetail.route)) {
+                        CustomVignette(
+                            visible = true,
+                            vignettePosition = VignettePosition.TopAndBottom
+                        )
+                    }
+                }
             },
             positionIndicator = {
-                CustomPositionIndicator(
-                    visible = scalingLazyListState.isScrollInProgress,
-                    scalingLazyListState = scalingLazyListState
-                )
+                // Only show position indicator for scrollable content.
+                currentRoute?.let {
+                    val activelyScrolling =
+                        scrollStates[currentScrollStateKey]?.isScrollInProgress ?: false
+
+                    if (currentRoute.startsWith(Screen.WatchList.route)) {
```

Comment:
This looks like it's simpler to just use the runtime type information to make the decision.  Or put this logic inside of CustomPositionIndicator so it takes a ScrollableState?

Reply:
Ah, good call, PositionIndictor takes care of this... maybe it didn't when I first wrote the sample? I can't remember. Either way, let me get rid of this and just use the stock PositionIndicator since it handles hiding itself.

Reply:
Fixed, PTAL

Reply:
I actually meant collapsing the current if {} else {} and getting the scrollState[currentId].  Then having an if else based on the type.

My thinking was this logic needs to change whenever routes change, while checking based on types you can even implement all 4 forms of PositionIndicator now and never change when routes changes.